tools/dav-client.cpp

```diff
@@ -0,0 +1,397 @@
+    {
+        if (reply.hasError()) {
+            qWarning() << reply.errorMessage;
+            qWarning() << reply.errorData;
+        } else {
+            qInfo() << "  create:";
+            qInfo() << "  - href:" << reply.uri;
+        }
+
+        execute("create-calendar");
+    }
+
+    void onCalendarUpdateFinished(const Buteo::Dav::Client::Reply &reply)
+    {
+        if (reply.hasError()) {
+            qWarning() << reply.errorMessage;
+            qWarning() << reply.errorData;
+        }
+
+        execute("update-calendar");
+    }
+
+    void onCalendarEtagsFinished(const Buteo::Dav::Client::Reply &reply,
+                                 const QHash<QString, QString> &etags)
+    {
+        if (reply.hasError()) {
+            qWarning() << reply.errorMessage;
+            qWarning() << reply.errorData;
+        } else {
+            qInfo() << "  etags:";
+            for (QHash<QString, QString>::ConstIterator it = etags.constBegin();
+                 it != etags.constEnd(); it++) {
+                qInfo() << "  - href:" << it.key();
+                qInfo() << "    etag:" << it.value();
+            }
+        }
+
+        execute("list-calendar-etags");
+    }
+
+    void onCalendarResourcesFinished(const Buteo::Dav::Client::Reply &reply,
+                                     const QList<Buteo::Dav::Resource> &resources)
+    {
+        if (reply.hasError()) {
+            qWarning() << reply.errorMessage;
+            qWarning() << reply.errorData;
+        } else {
+            qInfo() << "  resources:" << (resources.isEmpty() ? "[]" : "");
+            for (const Buteo::Dav::Resource &resource : resources) {
+                qInfo() << "  - href:" << resource.href;
+                qInfo() << "    etag:" << resource.etag;
+                qInfo() << "    status:" << resource.status;
+                qInfo() << "    data:" << resource.data;
+            }
+        }
+
+        execute("get-calendar-resources");
+    }
+
+    void onSendCalendarFinished(const Buteo::Dav::Client::Reply &reply, const QString &etag)
+    {
+        if (reply.hasError()) {
+            qWarning() << reply.errorMessage;
+            qWarning() << reply.errorData;
+        } else {
+            qInfo() << "  put:";
+            qInfo() << "  - href:" << reply.uri;
+            qInfo() << "  - etag:" << etag;
+        }
+
+        execute("p");
+    }
+
+    void onDeleteFinished(const Buteo::Dav::Client::Reply &reply)
+    {
+        if (reply.hasError()) {
+            qWarning() << reply.errorMessage;
+            qWarning() << reply.errorData;
+        } else {
+            qInfo() << "  delete:";
+            qInfo() << "  - href:" << reply.uri;
+        }
+
+        execute("d");
+    }
+
+    QCommandLineParser mParser;
+    Buteo::Dav::Client *mDAV;
+    QDateTime mFrom, mTo;
+};
+
+int main(int argc, char *argv[])
+{
+    DavCli app(argc, argv);
+    QCoreApplication::setApplicationName("dav-client");
```

Comment:
Pondering should this happen earlier or maybe inside the DavCli.

Reply:
Sure, I've now moved it in the DavCli constructor.